sdks/python/apache_beam/pvalue.py

```diff
@@ -319,9 +319,9 @@ def to_runner_api(self, context):
     return self._side_input_data().to_runner_api(context)
 
   @staticmethod
-  def from_runner_api(proto, context):
+  def from_runner_api(proto, coder):
```

Comment:
Given the coder comes from the PCollection and can be mutated, it's probably better to remove the coder from the side input object altogether, keeping the signature of this method as is, and instead passing it when we bind the PCollection.

Reply:
Seems this could also obviate the need for passing the transform proto itself in the PTransform.from_runner_api too.

Reply:
Looks much cleaner, trying this out now.